pdns/pdns_recursor.cc

```diff
@@ -3777,6 +3777,13 @@ static vector<pair<DNSName, uint16_t> >& operator+=(vector<pair<DNSName, uint16_
   return a;
 }
 
+static pair<int, string>& operator+=(std::pair<int, string>& a, const std::pair<int, string>& b)
+{
+  a.first |= b.first;
```

Comment:
I wonder if we should really do a OR here, because if we ever get two different non-zero values the result will not make any sense. Perhaps we should only override `a.first` if `a.first` is zero and `b.first` is non-zero?